### PR TITLE
Run tests on release branches

### PR DESCRIPTION
**Pull request by bcrochet**
Since we have release branches now, it is a good idea to run tests
on those branches so that when we cherry-pick to them, it will run
these tests.

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

**Review by acornett21 — APPROVED** [review]

File: .github/workflows/build-main.yml [hunk 1]
```diff
@@ -2,7 +2,10 @@ name: Build Main
 
 on:
   push:
-    branches: [ main, ghactions ]
+    branches:
+    - main
+    - ghactions
+    - release-*
```
Comment (itroyano) [inline .github/workflows/build-main.yml:8]: I'm not sure if the tagging here - alpha sha - would work for a release branch.

**Review by jomkz — APPROVED** [review]

**Comment by komish** [comment]
We probably also want to enable E2E tests for the release branch. 

Here's what we run against main:

https://github.com/openshift/release/blob/master/ci-operator/config/redhat-openshift-ecosystem/openshift-preflight/redhat-openshift-ecosystem-openshift-preflight-main__4.8.yaml

I can update this unless you want to take a stab at it.

**Comment by komish** [comment]
Also, do we want to run `build-main.yaml`? Do we want published images of the release branch?

**Comment by bcrochet** [comment]
> We probably also want to enable E2E tests for the release branch.
> 
> Here's what we run against main:
> 
> https://github.com/openshift/release/blob/master/ci-operator/config/redhat-openshift-ecosystem/openshift-preflight/redhat-openshift-ecosystem-openshift-preflight-main__4.8.yaml
> 
> I can update this unless you want to take a stab at it.

Yes we do. Go for it.

**Pull request merged**